ignition/service/messaging.py

```diff
@@ -210,6 +215,10 @@ def __lazy_init_producer(self):
             config['bootstrap_servers'] = self.bootstrap_servers
             config['client_id'] = 'ignition'
             self.producer = KafkaProducer(**config)
+    
+    def __close_producer(self):
+        self.producer.flush()
```

Comment:
Should check if `self.producer is not None` first in case it hasn't been initialised yet